x/mongo/driver/errors.go

```diff
@@ -170,15 +171,16 @@ func (wce WriteConcernError) NodeIsShuttingDown() bool {
 	return hasNoCode && strings.Contains(wce.Message, "node is shutting down")
 }
 
-// NotMaster returns true if this error is a not master error.
-func (wce WriteConcernError) NotMaster() bool {
-	for _, code := range notMasterCodes {
+// NotPrimary returns true if this error is a not primary error.
+func (wce WriteConcernError) NotPrimary() bool {
+	for _, code := range notPrimaryCodes {
 		if wce.Code == int64(code) {
 			return true
 		}
 	}
 	hasNoCode := wce.Code == 0
-	return hasNoCode && strings.Contains(wce.Message, "not master")
+	return hasNoCode &&
+		(strings.Contains(wce.Message, internal.LegacyNotPrimary) || strings.Contains(wce.Message, "not primary"))
```

Comment:
I do not think the check for the "not primary" message is required based on the [SDAM error handling pseudo-code](https://github.com/mongodb/specifications/blob/3da2c7816f6cdb449bd22603523c2dfc4b0af3e9/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#error-handling-pseudocode)

The error message checks are a backstop for old servers that do not contain the `code` field in errors. I do not think this will apply to 3.6+ servers that have the new "not primary" message. And this could subtly change behavior on old servers. For example: `db.runCommand({"not primary": 1})` can produce an error with the substring "not primary".

Aside: [DRIVERS-1152](https://jira.mongodb.org/browse/DRIVERS-1152) has additional recent background on error codes / message checks if you are interested.

Reply:
Good point. It seems like "not primary" should definitely be caught by the error code, so I've removed the redundant message check here and below.